tests/generate_libs/build.bp

```diff
@@ -130,11 +130,56 @@ bob_generate_binary {
     },
 }
 
+bob_generate_shared_library {
+    name: "libblah_shared_rename",
+    out: "libblah_shared2",
+    srcs: ["libblah/libblah.c"],
+    headers: [
+        "include/libblah.h",
+        "include/libblah_feature.h",
+    ],
+    export_gen_include_dirs: ["include"],
+    cmd: "{{.gen_cc}} -fPIC -o ${out} -shared ${in}; mkdir -p ${gen_dir}/include; cp ${module_dir}/libblah/libblah.h ${module_dir}/libblah/libblah_feature.h ${gen_dir}/include/.",
+    target: "host",
+    builder_android_bp: {
+        /* On Android BP generated libraries are not supported. */
+        enabled: false,
+    },
+
+}
+
+bob_binary {
+    name: "binary_linked_to_gen_shared_rename",
+    srcs: ["main.c"],
+    shared_libs: ["libblah_shared_rename"],
+    host_supported: true,
+    target_supported: false,
+    build_by_default: true,
+    builder_android_bp: {
+        /* On Android BP generated libraries are not supported. */
+        enabled: false,
+    },
+}
+
+bob_generate_source {
+    name: "check_renamed_gen_library",
+    out: ["success.txt"],
+    module_srcs: ["binary_linked_to_gen_shared_rename"],
+    /* Check that the binary has a NEEDED entry for the renamed shared library */
+    cmd: "{ objdump -p ${in} | grep NEEDED | grep -q libblah_shared2.so ; } && touch ${out}",
```

Comment:
Can we do this check in a python script to minimize the dependence on shell character handling?

Do we even need to do this check here? Why not just add a check to build_tests.sh to check for the presence of libblah_shared2.so?

Reply:
Sure. Although I think really we should be able to guarantee passing at least a small amount of syntax correctly to the shell, otherwise we end up with scripts for even the most trivial things.

The use case for this requires that the .so name and DT_NEEDED entries match the value of `out` - so I thought it would make the most sense to add it to the test.

Reply:
Done! Had to add another method (other than `objdump`) to get the tests working on OS X too.